src/app/modules/side-menu/components/region-filter/region-filter.component.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  CheckboxCustomEvent,
+  IonAccordion,
+  IonCheckbox,
+  IonItem,
+  IonList,
+  IonButton,
+  IonLabel,
+} from '@ionic/angular/standalone';
+import { ChangeDetectionStrategy, Component, Signal, computed, inject } from '@angular/core';
+import { NgIf } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
+import { distinctUntilChanged, EMPTY, map, Observable, shareReplay, switchMap, tap } from 'rxjs';
+import { SearchCriteriaService } from 'src/app/core/services/search-criteria/search-criteria.service';
+import { UserSettingService } from 'src/app/core/services/user-setting/user-setting.service';
+import { GeoHazard } from 'src/app/modules/common-core/models';
+import { LoggingService } from 'src/app/modules/shared/services/logging/logging.service';
+import { HeaderWithSelectedItemsComponent } from '../header-with-selected-items/header-with-selected-items.component';
+import { arrayHasNotChanged } from '../filter-menu/filter-menu.component';
+import { toSignal } from '@angular/core/rxjs-interop';
+
+const DEBUG_TAG = 'RegionFilterComponent';
+
+interface AvalancheRegion {
+  id: number;
+  name: string;
+  type: 'A' | 'B'; // A = regioner med fast varsling, B = regioner uten fast varsling
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  polygon: any; // TODO: Fix polygon;
+  checked?: boolean;
+}
+
+@Component({
+  selector: 'app-region-filter',
+  templateUrl: './region-filter.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [IonAccordion, IonCheckbox, IonItem, IonList, NgIf, HeaderWithSelectedItemsComponent],
+})
+/** Filtervalg for region / område */
+export class RegionFilterComponent {
+  private userSettingService = inject(UserSettingService);
+  private searchCriteriaService = inject(SearchCriteriaService);
+  private http = inject(HttpClient);
+  private logger = inject(LoggingService);
+  private currentGeoHazard = toSignal(this.userSettingService.currentGeoHazard$, { initialValue: [GeoHazard.Snow] });
+  isGeohazardSnow = computed(() => this.currentGeoHazard().includes(GeoHazard.Snow));
+
+  regions: Signal<AvalancheRegion[]>;
+  aRegions = computed(() => this.regions().filter((r) => r.type === 'A'));
+  bRegions = computed(() => this.regions().filter((r) => r.type === 'B')); // regioner uten fast varsling
+  selectedAregionNames = computed((): string[] => {
+    return this.regions()
+      .filter((r) => r.type === 'A')
+      .filter((r) => r.checked)
+      .map((r) => r.name);
+  });
+  selectedBregionNames = computed((): string[] => {
+    return this.regions()
+      .filter((r) => r.type === 'B')
+      .filter((r) => r.checked)
+      .map((r) => r.name);
+  });
+
+  constructor() {
+    const regions$ = this.userSettingService.currentGeoHazard$.pipe(
+      switchMap((geoHazards) => (geoHazards.includes(GeoHazard.Snow) ? this.getSnowRegions() : EMPTY)),
```

Comment:
Det er allerede en sjekk lengre ut for om geohazard er snø. Tror denne logikken kan forenkles litt, denne switchMap-en kan fjernes, og this.getSnowRegions brukes direkte?
Siden regions$ uansett brukes i en Signal, som jeg antar selv cacher resultatet, så kan trolig også shareReplay fjernes (men sjekk at regioner bare hentes én gang).